packages/app-project/stores/ProjectRoles.js

```diff
@@ -0,0 +1,61 @@
+import asyncStates from '@zooniverse/async-states'
+import { flow, getRoot, types } from 'mobx-state-tree'
+
+import numberString from './types/numberString'
+
+const ProjectRoles = types
+  .model('ProjectRoles', {
+    id: types.maybeNull(numberString),
+    loadingState: types.optional(types.enumeration('state', asyncStates.values), asyncStates.initialized),
+  })
+  .actions(self => {
+    const fetch = flow(function * fetch (projectId) {
+      const { client } = getRoot(self)
```

Comment:
This request only runs on the server, so shouldn’t be in the model. With Next 9.3 and up, server code can be split out of the client bundles by importing it directly into `getServerSideProps`. The pattern we’re using now is one where data is fetched in `getServerSideProps`/`getStaticProps` then passed to the browser in the `initialState` prop, which then hydrates the store in the browser. 

> Note: You can import modules in top-level scope for use in getServerSideProps. Imports used in getServerSideProps will not be bundled for the client-side.
>
>This means you can write server-side code directly in getServerSideProps. This includes reading from the filesystem or a database.

https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering

Reply:
hmm I see. @eatyourgreens what do you think about the fetch methods in the other models such as Project, YourStats, and Collections? I was using them as examples of how to make requests to panoptes. In the case of the Project model, if the request for project data happens only in getServerSideProps, when does the project data get attached to the model?

Reply:
The project model dates back to an older version of Next, when `getInitialProps` was used to fetch the data. If I open a PR to update it, would you be free to review it?

The project data itself gets applied to the model as a snapshot here:
https://github.com/zooniverse/front-end-monorepo/blob/0f7bff99885366a733fd6a3d0639c396a2bf45a0/packages/app-project/pages/_app.js#L28-L32
`initStore` runs on the server during the initial page render, then again in the browser when you first visit a project. After that, the browser should use client-side page transitions, where possible, so that store state isn't lost.

When I was setting that up (from the [NextJS MST example](https://github.com/vercel/next.js/tree/canary/examples/with-mobx-state-tree)) I found that the `initialState` snapshot constantly overwrites local store state in the browser, because we're using `getServerSideProps` to re-request the project data on every page transition. I worked around that by only applying the `project` and `ui` snapshots from the server data so be aware of that if you're adding any new models to the store. I don't think it will be a problem here, because you're adding models that are children of the project in the state tree.
https://github.com/zooniverse/front-end-monorepo/blob/0f7bff99885366a733fd6a3d0639c396a2bf45a0/packages/app-project/stores/initStore.js#L32-L41

Collections and YourStats run API requests in the browser, after the user has loaded, so I think it's fine to include those requests in the model. I guess an alternative would be to [use Suspense and local component state](https://mobx.js.org/react-integration.html#you-might-not-need-locally-observable-state) for those requests, but I don't think we need to worry about that right now.

The rough rule for the models, at the moment, is that an API request should run in the browser if it uses Auth headers ie. it requires a logged-in user.

Reply:
Sounds good @eatyourgreens. I'm happy to review any changes to the project model, and I think it would be helpful to see if the `getDefaultPageProps` and `getStaticPageProps` helpers change because I used them in the about pages similar to a project's home page.

I'll take a closer look at `/app-project/pages/_app.js` too, thank you for the explanation!

Reply:
#2052 should fetch the project data in `getStaticPageProps`, then pass it to the store in the `initialState` prop.

Reply:
We'll need auth headers for project requests to support private projects.

Reply:
Yep, but those won't be able to use SSR anyway.

Reply:
Maybe the project 404 page should log you in, then re-request the page with your Panoptes cookie? Then we could re-run the projects request but with credentials.

Reply:
Or we could actually build the server side authentication we discussed a while ago, however, this is out of scope of this PR I think. My point is that auth isn't necessarily a strict determiner of where requests should happen. I'd prefer we allow @goplayoutside3 to finish up the PR as is since she's soon going to be transitioning to the FF mapping visualizations project.

Reply:
Making the request with the Panoptes cookie would be server-side auth. My demo from a year ago, using the Panoptes session cookie, is here, but probably needs to be updated to the latest version of Next: https://github.com/zooniverse/next-cookie-auth-panoptes